spec/features/step_by_step_navigation_spec.rb

```diff
@@ -141,4 +162,8 @@ def and_I_see_my_selected_preferences
     expect(page.all(:select)[0].value).to eq("never")
     expect(page.all(:select)[1].value).to eq("always")
   end
+
+  def and_I_should_see_a_permission_exception
```

Comment:
The scheduling action should be hidden, rather than tell users that they don't have access to something. Also, making "scheduling" live will involve removing the scheduling permission, and moving access to scheduling to the normal "GDS Editor" permission. Think of the scheduling permission as a feature flag that we're going to turn off.

Reply:
Even so, we need to decide what should happen if an unauthorised user manages to get to the `/schedule` page, and have a test for that. Doesn't matter to me if it's an error message or a 403 response code or a redirect to the main step by step page. Any thoughts?

Reply:
Adding `require_scheduling_permissions!` to the controller will restrict access to users who have the "Scheduling" permissions. Other users who tried to hack the URL to access the schedule page will be kicked out. There is more information on how this works in the [gds-sso gem](https://github.com/alphagov/gds-sso/blob/b09c0510d417dce9f3a3fcad10630e29fd9c7a03/README.md#securing-your-application)

Reply:
Thank you for pointing that out. We were basically testing gds-sso gem. We are now testing if the ‘Schedule’ button doesn’t exist if you don’t have permission as well as 200 and 403 response in the controller.